### PR TITLE
Change KMP SDKs to external dependencies

### PR DESCRIPTION
**Pull request by wmontwe**
## Description

The project used a legacy Kotlin Multiplatform setup which prevents update to Kotlin 1.5. Also it made it hard to continue with the Java -> Kotlin conversion as it would require the whole project to be migrated at once, which is too much of an effort as old and new Kotlin project setups could not be mixed within one monorepo setup.

So the sub projects Auth, Crypto and Securestore are moved to external dependencies:

- [Data4Life Auth](https://github.com/d4l-data4life/hc-auth-sdk-kmp)
- [Data4Life Crypto](https://github.com/d4l-data4life/hc-crypto-sdk-kmp)
- [Data4Life Securestore](https://github.com/d4l-data4life/hc-securestore-sdk-kmp)

The new dependencies have been optimized and could be used in this project as composite builds and replace the maven dependency, see `settings.gradle.kts`. Just change the `gradle.properties` file and set for example `includeAuth` to `true`. This allows in project development of the mentioned dependencies.

The composite build Gradle feature has a limitation:  It requires matching Gradle Plugin versions. Which is currently not the case as this repository was already patched to newer ones and the dependencies not so far. Example Android Gradle Plugin here is 4.2.2 and in the dependencies 4.2.1

### Bumped

- Gradle 6.9 -> 6.9.1

### Known issues

- [x] FHIRUtils depends on fhir helper but that wasn't updated to the new utils and therefore the DataRestrictionException won't match -> solved by adding a patched copy

## Motivation and Context

Remove blocker to update to Kotlin 1.5 and enable conversion of Java to Kotlin on the core SDK, paving the path to convert to Kotlin Multiplatform.

## How is it being implemented?

The mentioned sub projects have been moved to it's own repositories and replaced within this project by maven dependencies.

## How Has This Been Tested?

The project was build and all test have been run successfully.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have updated the changelog accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: sample-android/build.gradle.kts [hunk 1]
```diff
@@ -58,7 +58,7 @@ android {
 
     buildTypes {
         getByName("debug") {
-            setMatchingFallbacks("release")
+            setMatchingFallbacks("debug", "release")
```
Comment (bitPogo) [inline sample-android/build.gradle.kts:61]: Does that mean we can get rid of `Debug` in the releases?

File: sample-android/src/main/java/care/data4life/sdk/helpers/stu3/AttachmentBuilderPatched.kt [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+import care.data4life.fhir.stu3.model.Attachment
+import care.data4life.fhir.stu3.model.FhirDateTime
+import care.data4life.sdk.config.DataRestriction
+import care.data4life.sdk.lang.DataRestrictionException
+import care.data4life.sdk.util.Base64
+import care.data4life.sdk.util.HashUtil
+import care.data4life.sdk.util.MimeType
+
+object AttachmentBuilderPatched {
```
Comment (bitPogo) [inline sample-android/src/main/java/care/data4life/sdk/helpers/stu3/AttachmentBuilderPatched.kt:27]: This is due to the fact, that the FHIR-helpers are still one version behind right?

**Review by bitPogo — COMMENTED** [review]
Just 1 nitpick and 2 questions.

Also please check the former crypto module - I think there is a leftover.

File: CHANGELOG.adoc [hunk 3]
```diff
@@ -44,6 +50,8 @@ toc::[]
 === Bumped
 
 * mockk 1.10.0 -> 1.10.6
+* Gradle 6.8.3 -> 6.9
+* HC Util SDK 1.6.0 -> 1.8.0
```
[the remark this answers is not in the document]
Comment (bitPogo) [reply CHANGELOG.adoc:54]: Also it is in the wrong place.

File: sample-android/src/main/java/care/data4life/sdk/helpers/stu3/AttachmentBuilderPatched.kt (hunk 2 above)
Comment (wmontwe) [reply sample-android/src/main/java/care/data4life/sdk/helpers/stu3/AttachmentBuilderPatched.kt:27]: They are on an incompatible version of the utils, next time such refactorings happen they need to be marked as breaking.

File: sample-android/build.gradle.kts (hunk 1 above)
Comment (wmontwe) [reply sample-android/build.gradle.kts:61]: we shouldn't need debug for `release`, but for `debug` we might need `release` as fallback as not all dependencies have been published with the right dependencies

File: CHANGELOG.adoc (hunk 3 above)
[the remark this answers is not in the document]
Comment (wmontwe) [reply CHANGELOG.adoc:54]: that came through the merge, apparently

**Comment by wmontwe** [comment]
> Just 1 nitpick and 2 questions.
> 
> Also please check the former crypto module - I think there is a leftover.

The last merge of main introduced patches to the crypto. Could be deleted

**Review by bitPogo — APPROVED** [review]

**Pull request merged**